superset/models/core.py

```diff
@@ -368,14 +372,41 @@ def get_sqla_engine_with_context(
         schema: Optional[str] = None,
         nullpool: bool = True,
         source: Optional[utils.QuerySource] = None,
+        ssh_tunnel: Optional["SSHTunnel"] = None,
     ) -> Engine:
-        yield self._get_sqla_engine(schema=schema, nullpool=nullpool, source=source)
+        ssh_params = {}
+        if ssh_tunnel:
+            # build with override
+            url = make_url_safe(self.sqlalchemy_uri_decrypted)
+            ssh_tunnel.bind_host = url.host
```

Comment:
Shouldn't this be dynamic? Doesn't seem to me that this would work if you entered some arbitrary value in the UI for `bind_host`, for example.

Reply:
the bind_host and bind_port is coming from the sqla uri representing where on the VPC the database is available